bin/git-create-branch

```diff
@@ -1,8 +1,35 @@
 #!/usr/bin/env bash
 
-branch=$1
-test -z $branch && echo "branch required." 1>&2 && exit 1
+for param in $*
+do
+	case $param in
+		-r|--remote)
+			if [ $# -eq 2 ]
+			then
+				BRANCH=$2
+			else			
+				REMOTE=$2
+				BRANCH=$3
+			fi
+			;;
+		*)
+			BRANCH=$param
+	esac
+done
+
+test -z $BRANCH && echo "branch required." 1>&2 && exit 1
+if [[ -n $REMOTE ]]
+then
+	git ls-remote --exit-code $REMOTE &>/dev/null
+	if [ $? = 0 ]
+	then
+		git push $REMOTE HEAD:refs/heads/$BRANCH
+		git fetch $REMOTE
+		git checkout --track -b $BRANCH $REMOTE/$BRANCH
```

Comment:
IMHO, if we could add `exit $?` after here, the code structure would be more concise.